test/framework/src/main/java/org/elasticsearch/test/InternalAggregationTestCase.java

```diff
@@ -317,13 +325,28 @@ public final void testFromXContent() throws IOException {
         } else {
             originalBytes = toXContent(aggregation, xContentType, params, humanReadable);
         }
+        BytesReference mutated;
+        if (addRandomFields) {
+            /**
+             * - we don't add to the root object because it should only contain the named aggregation to test
```

Comment:
For the aggregation tests, we render the test item renders as `{ "someType#xyz : { .... }  }` and we only really need to check that the inner object parses back correctly.